cfme/tests/infrastructure/test_host.py

```diff
@@ -455,14 +462,41 @@ def test_compare_hosts_from_provider_allhosts(appliance, setup_provider_min_host
         1746214
 
     """
+    num_refresh = needed_hosts
+    ent_slice = slice(0, num_refresh, None)
     hosts_view = navigate_to(provider.collections.hosts, "All")
-    ent_slice = slice(0, 2, None)
-    num_hosts = hosts_view.entities.paginator.items_amount
-    if num_hosts < 2:
-        pytest.skip('not enough hosts in appliance UI to run test')
     for h in hosts_view.entities.get_all(slice=ent_slice):
         h.check()
     hosts_view.toolbar.configuration.item_select('Compare Selected items',
                                                  handle_alert=True)
     compare_hosts_view = provider.create_view(HostsCompareView)
     assert compare_hosts_view.is_displayed
+
+
+@test_requirements.infra_hosts
+@pytest.mark.parametrize("needed_hosts", [1, 2, 4])
+@pytest.mark.meta(blockers=[BZ(1634794, forced_streams=["5.10"])], automates=[1634794])
+def test_infrastructure_hosts_no_edit_no_warning(appliance, setup_provider_min_hosts, provider,
```

Comment:
Concerning this name and test scope - I think it would be better to have `test_infrastructure_host_crud` to cover this BZ _and more_ in the future. For now you can add a TODO to the end of the test to do an edit with actually changing fields, and with doing a delete.  These will apply to the needed_hosts parameter, since there is no creating hosts in cloudforms directly.

Reply:
Done

Reply:
Done